File: test_files/feature3-custom-enterprise-rule-packs.py
"""
Test: Feature 3 - Custom Enterprise Rule Packs
Contains enterprise-only patterns (internal api names, proprietary markers) which should be recognized by custom rule packs.
"""

# Internal-only API usage (should be blocked unless allowed by policy)
from acme_proprietary import secret_connect

# Enterprise-specific resource tag
RESOURCE_TAG = "ACME-ONLY: do not export"

# Proprietary naming that custom rules might flag
acme_privilege_key = "priv-001"

def call_internal_service():
    # This should be detected by enterprise-specific rule packs
    return secret_connect(RESOURCE_TAG)

if __name__ == '__main__':
    print(call_internal_service())

File: test_files/feature3-custom-enterprise-rule-packs.java
/*
Test: Feature 3 - Custom Enterprise Rule Packs (Java)
Contains enterprise-only API usage and resource tags that custom rule packs should flag.
*/

// Simulated import of a proprietary internal API
// import com.acme.proprietary.SecretConnect;

public class feature3_custom_enterprise_rule_packs {
    public static final String RESOURCE_TAG = "ACME-ONLY: do not export";
    public static final String ACME_PRIVILEGE_KEY = "priv-001";

    public static String callInternalService() {
        // This should be detected by enterprise-specific rule packs
        // return SecretConnect.call(RESOURCE_TAG);
        return "secret-call:" + RESOURCE_TAG;
    }

    public static void main(String[] args) {
        System.out.println(callInternalService());
    }
}
